sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobAsyncClient.java

```diff
@@ -374,6 +377,76 @@ public void getBlobProperties(String containerName,
+     * @param headers {@link BlobHttpHeaders}
+     * @param callback      Callback that receives the response.
+     */
+    public void setHttpHeaders(String containerName,
+                               String blobName,
+                               BlobHttpHeaders headers,
+                               CallbackWithHeader<Void, BlobSetHttpHeadersHeaders> callback) {
+        storageBlobServiceClient.setBlobHttpHeaders(containerName, blobName, headers, callback);
+    }
+
+    /**
+     * Changes a blob's HTTP header properties. If only one HTTP header is updated, the others will all be erased. In
+     * order to preserve existing values, they must be passed alongside the header being changed.
+     *
+     * @param containerName The container name.
+     * @param blobName      The blob name.
+     * @param timeout           The timeout parameter is expressed in seconds. For more information, see
+     *                          &lt;a href="https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/setting-timeouts-for-blob-service-operations"&gt;Setting Timeouts for Blob Service Operations.&lt;/a&gt;.
+     * @param version               Specifies the version of the operation to use for this request.
+     * @param leaseId           If specified, the operation only succeeds if the resource's lease is active and
+     *                          matches this ID.
+     * @param ifModifiedSince   Specify this header value to operate only on a blob if it has been modified since the
+     *                          specified date/time.
+     * @param ifUnmodifiedSince Specify this header value to operate only on a blob if it has not been modified since
+     *                          the specified date/time.
+     * @param ifMatch           Specify an ETag value to operate only on blobs with a matching value.
+     * @param ifNoneMatch       Specify an ETag value to operate only on blobs without a matching value.
+     * @param ifTags            Specify a SQL statement to apply to the tags on the blob.
+     * @param headers           {@link BlobHttpHeaders}
+     * @param requestId         Provides a client-generated, opaque value with a 1 KB character limit that is
+     *                          recorded in the analytics logs when storage analytics logging is enabled.
+     * @param cancellationToken The token to request cancellation.
+     * @param callback      Callback that receives the response.
+     */
+    public void setHttpHeadersWithResponse(String containerName,
+                                                                 String blobName,
+                                                                 Integer timeout,
+                                                                 String version,
+                                                                 String leaseId,
+                                                                 DateTimeRfc1123 ifModifiedSince,
```

Comment:
Should we be taking OffsetDateTime here instead?

Reply:
We could just pass this in as BlobRequestConditions

Reply:
That's what I ended up doing. I didn't notice they were already using it